etc/consular/nginx-services.ctmpl

```diff
@@ -0,0 +1,27 @@
+# NOTE: This is an nginx configuration automatically generated by
+#       consul-template, there's no point in making manual changes
+#       to this file.
+#
+#       This expects a key to exist under "consular/<app_id>/location"
+#       in Consul's KV store. This is used to map a path to a service in
+#       Nginx. If the key does not exist the service isn't added to the
+#       list of services in the Nginx config.
+
+{{range services}}{{$labels := ls (print "consular/" .Name) | explode}}{{if $labels.location}}
+upstream {{.Name}} { {{range service .Name}}
+    server {{.Address}}:{{.Port}};{{end}}
+}
+{{end}}{{else}}
+# Skipped service {{.Name}} as it does not have a KV domain entry.
+{{end}}
+server {
+    listen 80;
```

Comment:
We probably want to mark this as [`internal`](http://nginx.org/en/docs/http/ngx_http_core_module.html#internal)